make/photon/db/docker-entrypoint.sh

```diff
@@ -1,120 +1,59 @@
 #!/bin/bash
 set -e
 
-# usage: file_env VAR [DEFAULT]
-#    ie: file_env 'XYZ_DB_PASSWORD' 'example'
-# (will allow for "$XYZ_DB_PASSWORD_FILE" to fill in the value of
-#  "$XYZ_DB_PASSWORD" from a file, especially for Docker's secrets feature)
-file_env() {
-        local var="$1"
-        local fileVar="${var}_FILE"
-        local def="${2:-}"
-        if [ "${!var:-}" ] && [ "${!fileVar:-}" ]; then
-                echo >&2 "error: both $var and $fileVar are set (but are exclusive)"
-                exit 1
-        fi
-        local val="$def"
-        if [ "${!var:-}" ]; then
-                val="${!var}"
-        elif [ "${!fileVar:-}" ]; then
-                val="$(< "${!fileVar}")"
-        fi
-        export "$var"="$val"
-        unset "$fileVar"
-}
-
-# look specifically for PG_VERSION, as it is expected in the DB dir
-if [ ! -s "$PGDATA/PG_VERSION" ]; then
-        file_env 'POSTGRES_INITDB_ARGS'
-        if [ "$POSTGRES_INITDB_XLOGDIR" ]; then
-                export POSTGRES_INITDB_ARGS="$POSTGRES_INITDB_ARGS --xlogdir $POSTGRES_INITDB_XLOGDIR"
-        fi
-        initdb -D $PGDATA  -U postgres -E UTF-8 --lc-collate=en_US.UTF-8 --lc-ctype=en_US.UTF-8 $POSTGRES_INITDB_ARGS
-        # check password first so we can output the warning before postgres
-        # messes it up
-        file_env 'POSTGRES_PASSWORD'
-        if [ "$POSTGRES_PASSWORD" ]; then
-                pass="PASSWORD '$POSTGRES_PASSWORD'"
-                authMethod=md5
-        else
-                # The - option suppresses leading tabs but *not* spaces. :)
-                cat >&2 <<-EOF
-                        ****************************************************
-                        WARNING: No password has been set for the database.
-                                        This will allow anyone with access to the
-                                        Postgres port to access your database. In
-                                        Docker's default configuration, this is
-                                        effectively any other container on the same
-                                        system.
-                                        Use "-e POSTGRES_PASSWORD=password" to set
-                                        it in "docker run".
-                        ****************************************************
-EOF
-
-                pass=
-                authMethod=trust
+source $PWD/initdb.sh
+
+CUR=$PWD
+TO_UPGRADE=false
+PG_VERSION_NEW=$1
+
+PGDATAOLD=""
+PGDATANEW=${PGDATA}/pg${PG_VERSION_NEW}
+
+#
+# Init DB: $PGDATA is empty.
+# Upgrade DB: 1, has $PGDATA\PG_VERSION. 2, has pg version directory with PG_VERSION inside.
+# Error: $PGDATA is not empty and has files or invalid directory.
+#
+if [ "$(ls -A $PGDATA)" ]; then
+        if [ ! -d $PGDATANEW ]; then
+                TO_UPGRADE=true
+                if [ -s $PGDATA/PG_VERSION ]; then
```

Comment:
I think you can search for the file `PG_VERSION` under $PGDATA, and use the `dirname` of the file as `$PGDATAOLD`, such that you don't need the `if...else...` in line 21~32

Reply:
updated.